webview/src/experiments/components/table/header/TableHead.tsx

```diff
@@ -44,6 +44,7 @@ export const TableHead = ({
   )
   const dispatch = useDispatch()
   const orderedColumns = useColumnOrder(columns, columnOrder)
+  let leave_enter_counter = 0
```

Comment:
Here is the main change - logic around this counter, and `onDragLeave` below

Reply:
yes, good catch .. using multiple languages simultaneously hurt me

Reply:
There is already the same functionality is `useDeferedDragLeave` with the boolean `hoveringSomething`. Also we try to avoid using `let` variables inside of React. We use `useState` or `useRef` instead. In this case since we don't need a re-render we'd use a ref. That will avoid side effects when mouting/unmounting the component.

Reply:
Thanks @sroy3 . This is useful. `useDeferedDragLeave` looks good (this case is a bit different, but I hope it'll work the same w/o any flickering). Two questions - I see that it has 500 as a delay, can we make it smaller? Any specific reason for this?

Also, I would need to call `dispatch` on an actual, final leave. What would be the best way? Parametrize and pass a callback or do this on the component level - some hook that depends on `hoveringSomething`? What would be the best?

Reply:
You can use `immediateDragLeave` and `immediateDragEnter` instead (inside the same file). If I recall correctly the delay is used because of entering/leaving groups vs entering/leaving another plot.

Reply:
sorry, to clarify - why is it 500, not 50 or 10 for example (I think delay is used for a few reasons there - to. prevent flickering - e.g. left one plot -> entered another - in between those actions we don't want to react, also in some browsers events are generated in a different order). So, I'm fine with defered logic - that's exactly what I need I think. I want to make it faster though.

Let me do this, and I'll ask you to review.

Reply:
Lowering the timeout does not seem to have any negative effect. I can't say with 100% certitude, but I think 500 might have been chosen practically at random.